src/navigation/gate.py

```diff
@@ -70,11 +70,49 @@ def spider_gate_trajectory(cls, approach_distance: float, gate: Gate, rover_posi
         closest_approach_point = possible_approach_points[approach_idx]
         victory_point = possible_approach_points[1 - approach_idx]
 
+        coordinates = GateTrajectory.gateSelectPath(
+            rover_position, closest_prep_point, closest_approach_point, center, victory_point, gate
+        )
+
         # put the list of coordinates together
-        coordinates = np.array([closest_prep_point, closest_approach_point, victory_point])
-        coordinates = np.hstack((coordinates, np.zeros(coordinates.shape[0]).reshape(-1, 1)))
         return GateTrajectory(coordinates)
 
+    def gateSelectPath(
+        rover_position: np.ndarray,
+        prep: np.ndarray,
+        approach: np.ndarray,
+        center: np.ndarray,
+        done: np.ndarray,
+        gate: Gate,
+    ):
+        # Returns the path we will use to traverse the gate, using as few approach points as possible
+
+        # Get the shapes of both the posts
+        postOneShape, postTwoShape = gate.getPostShape()
+
+        rover = np.array([rover_position[0], rover_position[1]])
+
+        # try paths with successively more points until we have one that won't intersect
+        allPts = [prep, approach, center, done]
+        start_index = 2
+        path = makeShapelyPath(rover, allPts[start_index:])
+        while path.intersects(postOneShape) or path.intersects(postTwoShape):
+            start_index -= 1
+            if start_index == 0:
+                break
+            path = makeShapelyPath(rover, allPts[start_index:])
+
+        coordinates = np.array(allPts[start_index:])
+        coordinates = np.hstack((coordinates, np.zeros(coordinates.shape[0]).reshape(-1, 1)))
+        return coordinates
+
+
+def makeShapelyPath(rover, pathPts):
+    # makes a path we can use to check intersection
+    pathList = [list(rover)]
```

Comment:
turns out shapely does integrate with numpy pretty well, so the `LineString` constructor can take in a 2D numpy array instead of a nested python list. Since all of our data is already in the form of numpy arrays, it probably makes sense to keep it that way.

 There are a few ways you could do this, my suggestion would be to make `allPts` a 2D array by doing `np.vstack((prep, approach, center, done))`. Then you can index into it like `allPts[start_index:, :]`, and then in this makeShapelyPath function, which will take in a 1D array for rover and a 2D array for pathPts, you can just `np.vstack((rover, pathPts))` and then feed that into the LineString constructor.